qualtran/bloqs/and_bloq.py

```diff
@@ -291,6 +294,9 @@ def decompose_from_registers(
         )
         yield self._decompose_via_tree(control, self.cvs, ancilla, *target)
 
+    def decompose_bloq(self) -> 'CompositeBloq':
+        return decompose_from_cirq_style_method(self)
```

Comment:
@tanujkhattar the idea is to explicitly flag that we're using a cirq style `decompose_from_registers` method by overriding `decompose_bloq` with this non-default way of doing a decomposition. You can actually name the decomposition method whatever you want and specify its name as the second argument. There's a variety of alternatives centering around a) plumbing through the method b) defaults.

I picked this way to keep the `Bloq` abstract interface streamlined, with the happy path clearly highlighted. The explicit delegation with `decompose_from_cirq_style_method` removes a category of headaches when debugging a decomposition.